pkg/providers/common/auditpolicy.go

```diff
@@ -0,0 +1,252 @@
+package common
+
+import (
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	"sigs.k8s.io/yaml"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+func GetAuditPolicyYaml(kubeVersion anywherev1.KubernetesVersion) ([]byte, error) {
+	return yaml.Marshal(GetAuditPolicy(kubeVersion))
+}
+
+func GetAuditPolicy(kubeVersion anywherev1.KubernetesVersion) *auditv1.Policy {
+	a := baseAuditPolicy()
+	if kubeVersion == anywherev1.Kube124 {
+		a.APIVersion = "audit.k8s.io/v1"
+	} else {
+		a.APIVersion = "audit.k8s.io/v1beta1"
+	}
+
+	return a
+}
+
+func baseAuditPolicy() *auditv1.Policy {
```

Comment:
when this gets serialized, do we have any way to validate that it won't have any deltas compared to what was being generated with the yaml spec? Otherwise we'll still have a rolling upgrade, but just due to serialization differences and the way the yaml is being generated.